test/src/e2e_vm_tests/mod.rs

```diff
@@ -124,6 +126,8 @@ pub fn run(filter_regex: Option<regex::Regex>) {
         "star_import_alias",
         "item_used_without_import",
         "shadow_import",
+        "supertrait_dup_methods_1",
+        "supertrait_dup_methods_2",
     ];
```

Comment:
The asserting tests are good -- one more test though: do you have one which will fail compilation if we try to impl  a subtrait for a type which is missing the required supertrait?

Reply:
Do you mean something like this? 
```rust
script;

trait Type {
    fn get_number_of_bytes(self) -> u64;
}

trait Numeric {} // Missing `: Type`

struct U64 {
    n: u64,
}

impl Numeric for U64 {
    fn get_number_of_bytes(self) -> u64 {
        8
    }
}
```
Yes this fails with 
```
error
  --> test/src/main.sw:14:8
   |
12 | 
13 | impl Numeric for U64 {
14 |     fn get_number_of_bytes(self) -> u64 {
   |        ^^^^^^^^^^^^^^^^^^^ Function "get_number_of_bytes" is not a part of trait "Numeric"'s interface surface.
15 |         8
16 |     }
   |
error
  --> test/src/main.sw:28:20
   |
26 | 
27 | //    let b = Bool { b : true };
28 |     let bytes1 = n.get_number_of_bytes();
   |                    ^^^^^^^^^^^^^^^^^^^ No method named "get_number_of_bytes" found for type "struct U64<u64>".
29 | //    let bytes2 = b.get_number_of_bytes();
30 | }
   |
  Aborting due to 2 errors.
```
but passes when I add back `: Type` in the definition of `trait Numeric`.

Reply:
This is great, but I was actually also thinking more around:
```rust
trait A {
  fn a();
}
trait B: A {
  fn b();
}
struct X { ... };
impl B for X {
  fn b() { ... }
}
```
It should complain about `A` not being implemented for X, shouldn't it?  Something like:
```
   |
14 | impl B for X {
   |      ^ the trait `A` is not implemented for `X`
   |
note: required by a bound in `B`
  --> src/main.rs:10:10
   |
10 | trait B: A {
   |          ^ required by this bound in `B`
```

Sorry to keep niggling on this. :slightly_smiling_face:

Reply:
Ahhh yes that does fail to compile with the error you just described. Let me add a test right now :)

Reply:
Added the test. The error message is as follows:
```
$ forc build
error
  --> missing_supertrait_impl/src/main.sw:13:1
   |
11 |   
12 |   
13 | / impl B for X {
14 | |     fn b() { }
15 | | // This code shouldn't compile because the implementation of `a()` below is missing.
16 | | //    fn a() { } 
17 | | }
   | |_^ Functions are missing from this trait implementation: a
18 |   
19 |   fn main() { } 
   |
  Aborting due to 1 error.
Error: "Failed to compile missing_supertrait_impl"
```